### PR TITLE
Add to LoggingError a property bag to hold telemetry props

### LINKED ISSUE #6485: LoggingError should hold telemetry props in a property rather than logging all its properties
In other words, we want it to be that `LoggingError` "has a" property bag for logging, rather than `LoggingError` "is a" property bag for logging.

Couple examples of where the current approach is troublesome:
* https://github.com/microsoft/FluidFramework/pull/5790 where a property (`claims`) needed at runtime shouldn't be logged.
* The `__isFluidLoggingError__` property shows up in our logs (this is also a weak way to check ID, see https://github.com/microsoft/FluidFramework/issues/6468)
* #6400 where `IGenericError` has an `error` member that you might think would be logged but won't be (phew!)
* #6008 where runtime code (parsing the error response stored on the error obj) depended on the telemetry props being _on_ the object - also making it very hard to track back to where the property was even set.

The general statement is that the shape of the object at runtime should be decoupled from the set of props logged.  I'm also concerned about the current approach leading people to make bad choices like casting to `any` to set an arbitrary logging property - and then maybe elsewhere someone starts reading that property.  Then we would have runtime behavior not properly modeled by the type system.

** Proposed Fix **

The reality is that there may be runtime dependencies on telemetry props existing on the object - This I am unwilling to break.  There may also be cases where someone's used to seeing a property in the logs that's set directly on the error object in the code (not through `addTelemetryProperties`) - This I _am_ willing to break, after doing some due diligence to convert places that do set properties for logging.

So in the end, `addTelemetryProperties` will add the props to a property bag, _as well as_ set them on the object itself.  And I will also make a pass over every class extending `LoggingError` to add the appropriate class members to the property bag.

### PR DESCRIPTION
**Pull request by markfields**
Fixes #6485 

Need to add/update tests.

### CODE REVIEW

File: packages/utils/telemetry-utils/src/logger.ts [hunk 1]
```diff
@@ -578,8 +582,15 @@ export class LoggingError extends Error implements ILoggingError {
     constructor(
         message: string,
         props?: ITelemetryProperties,
+        omitPropsFromLogging: string[] = [],
```
Comment (anthony-murphy) [inline packages/utils/telemetry-utils/src/logger.ts:585]: should this be a set, rather than an array

**Comment by anthony-murphy** [comment]
title doesn't match the PR

File: packages/loader/driver-utils/src/network.ts [hunk 2]
```diff
@@ -71,7 +71,8 @@ export class AuthorizationError extends LoggingError implements IAuthorizationEr
         readonly tenantId: string | undefined,
         props?: ITelemetryProperties,
     ) {
-        super(errorMessage, props);
+        // don't log claims or tenantId
+        super(errorMessage, props, ["claims", "tenantId"]);
```
Comment (anthony-murphy) [inline packages/loader/driver-utils/src/network.ts:75]: this is a bit strange to me, shouldn't downstream users be tagging?

**Comment by markfields** [comment]
Rethinking this, and needs to get some merges fixed up from recent related changes.  May reopen.

**Pull request closed without merging**